clients/backend.go

```diff
@@ -39,9 +41,27 @@ func NewBackendFactory(c config.Config) *BackendFactory {
 	}
 }
 
-func (f BackendFactory) NewClient(route *models.Route) *http.Client {
-	return &http.Client{
+func (f BackendFactory) NewClient(route *models.Route, followRedirect bool) *http.Client {
+	client := &http.Client{
 		Transport: f.factory.New(route.ServerCertDomainSan),
 		Timeout:   30 * time.Second,
 	}
+
+	if ( ! followRedirect ) {
+		client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+			if ( len(via) == 0 ) {
+				return errors.New(fmt.Sprintf("empty previous request for redirect %s, should not happen !", req.URL))
+			}
+
+			if (req.URL.Host == via[0].URL.Host) {
```

Comment:
as discussed, we give here, a chance to follow a legitimate redirect (relative or same host as previous)